### DISCUSSION TITLE
Reference for current diffraction implementation

### DISCUSSION THREAD

**Discussion by jeertmans** [NVlabs/sionna#205] [Q&A]
Hello, when reading your source code, I do not fully understand how you find the interaction point for diffraction. More precisely, this part of the code is a bit cryptic to me.
Do you have any reference explaining where this algorithm or method originates from?

Thank you in advance!

https://github.com/NVlabs/sionna/blob/f5ea373b948bfc367eaccdb8907889cfe62badd9/sionna/rt/solver_paths.py#L2368-L2437

**Comment by faycalaa** [comment]
Hello,

We are in the process of developing comprehensive documentation for the path tracing algorithms. However, it's important to note that this will require some time to complete.

**Reply by jeertmans** [reply]
Hello, no problems, thank you for your answer and for this great tool!

**Comment by gauravduggal** [comment]
The location of the diffraction point can be computed using the diffraction law or fermat's principle of least time. This paper has the related math: https://ieeexplore.ieee.org/abstract/document/10879253
